File: Back-end/src/main/java/KJX/KJX/entity/Schedule.java
package KJX.KJX.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "schedules", indexes = {
    @Index(name = "idx_schedules_team_id", columnList = "team_id"),
    @Index(name = "idx_schedules_day", columnList = "day_of_week")
})
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Schedule {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id")
    private Team team;
    
    @Column(name = "day_of_week", nullable = false, length = 10)
    private String dayOfWeek; // "LUNDI", "MARDI", etc.
    
    @Column(name = "start_time", length = 10)
    private String startTime; // "10:00"
    
    @Column(name = "end_time", length = 10)
    private String endTime; // "13:00"
    
    @Column(length = 255)
    private String activity; // "Entraînement", "Scrim", "Repos", etc.
    
    @Column(columnDefinition = "NVARCHAR(MAX)")
    private String notes;
    
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: nodejs-backend/src/models/Schedule.js
/**
 * Schedule Model
 * Converted from Spring Boot JPA Entity
 */

const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const Schedule = sequelize.define('Schedule', {
    id: {
      type: DataTypes.BIGINT,
      primaryKey: true,
      autoIncrement: true
    },
    team_id: {
      type: DataTypes.BIGINT,
      field: 'team_id',
      references: {
        model: 'teams',
        key: 'id'
      }
    },
    day_of_week: {
      type: DataTypes.STRING(10),
      allowNull: false,
      field: 'day_of_week'
    },
    start_time: {
      type: DataTypes.STRING(10),
      field: 'start_time'
    },
    end_time: {
      type: DataTypes.STRING(10),
      field: 'end_time'
    },
    activity: {
      type: DataTypes.STRING(255)
    },
    notes: {
      type: DataTypes.TEXT
    },
    created_at: {
      type: DataTypes.DATE,
      defaultValue: DataTypes.NOW,
      field: 'created_at'
    },
    updated_at: {
      type: DataTypes.DATE,
      defaultValue: DataTypes.NOW,
      field: 'updated_at'
    }
  }, {
    tableName: 'schedules',
    timestamps: true,
    createdAt: 'created_at',
    updatedAt: 'updated_at',
    indexes: [
      {
        fields: ['team_id'],
        name: 'idx_schedules_team_id'
      },
      {
        fields: ['day_of_week'],
        name: 'idx_schedules_day'
      }
    ]
  });

  return Schedule;
};
